pkg/image/admission/imagepolicy/api/v1/types.go

```diff
@@ -0,0 +1,138 @@
+	"k8s.io/kubernetes/pkg/api/v1"
+)
+
+// ImagePolicyConfig is the configuration for control of images running on the platform.
+type ImagePolicyConfig struct {
+	unversioned.TypeMeta `json:",inline"`
+
+	// ExecutionRules determine whether the use of an image is allowed in an object with a pod spec.
+	// By default, these rules only apply to pods, but may be extended to other resource types.
+	// If all execution rules are negations, the default behavior is allow all. If any execution rule
+	// is an allow, the default behavior is to reject all.
+	ExecutionRules []ImageExecutionPolicyRule `json:"executionRules"`
+
+	// TODO: enable additional rule types
+	// ConsumptionRules are applied when creating resources with pod specs.
+	// ConsumptionRules []ImageConsumptionPolicyRule `json:"consumptionRules"`
+	// PlacementRules are applied when creating resources with pod specs.
+	// PlacementRules []ImagePlacementPolicyRule `json:"placementRules"`
+}
+
+// ImageExecutionPolicyRule determines whether a provided image may be used on the platform.
+type ImageExecutionPolicyRule struct {
+	ImageCondition `json:",inline"`
+
+	// Resolve indicates that images referenced by this resource must be resolved
+	Resolve bool `json:"resolve"`
+}
+
+// ImageCondition defines the conditions for matching a particular image source. The conditions below
+// are all required (logical AND). If Reject is specified, the condition is false if all conditions match,
+// and true otherwise.
+type ImageCondition struct {
+	// Name is the name of this policy rule for reference. It must be unique across all rules.
+	Name string `json:"name"`
+	// IgnoreNamespaceOverride prevents this condition from being overriden when the
+	// `alpha.image.policy.openshift.io/ignore-rules` is set on a namespace and contains this rule name.
+	IgnoreNamespaceOverride bool `json:"ignoreNamespaceOverride"`
+
+	// Reject indicates this rule is inverted - if all conditions below match, then this rule is considered to
+	// not match.
+	Reject bool `json:"reject"`
+
+	// OnResources determines which resources this applies to. Defaults to 'pods' for ImageExecutionPolicyRules.
+	OnResources []unversioned.GroupResource `json:"onResources"`
+
+	// MatchIntegratedRegistry will only match image sources that originate from the configured integrated
+	// registry.
+	MatchIntegratedRegistry bool `json:"matchIntegratedRegistry"`
+	// MatchRegistries will match image references that point to the provided registries
+	MatchRegistries []string `json:"matchRegistries"`
+
+	// AllowResolutionFailure allows the subsequent conditions to be bypassed if the integrated registry does
+	// not have access to image metadata (no image exists matching the image digest).
+	AllowResolutionFailure bool `json:"allowResolutionFailure"`
+
+	// MatchDockerImageLabels checks against the resolved image for the presence of a Docker label
+	MatchDockerImageLabels []ValueCondition `json:"matchDockerImageLabels"`
```

Comment:
I want this in because it's going to overlap in function and I don't see the point of removing it.  I don't think it's just consumption - in locked down environments I can easily imagine this being used to control production as "safety" or as a temporary blocker.